Adds a Flask Guide, attempt 2 (#3454)
platformsh/platformsh-docs: sites/friday/src/get-started/flask/web-servers.md

---
title: Switching from the Flask server to an alternative Web Server
sidebarTitle: Change web server
description: Steps for changing the web server on {{% vendor/name %}}.
---

> While lightweight and easy to use, Flask’s built-in server is not suitable for production as it doesn’t
> scale well. {{% vendor/name %}} [supports several different web servers](/languages/python/server/_index.md) for Python

Reviewer: ```suggestion
While lightweight and easy to use, Flask’s built-in server is not suitable for production as it doesn’t
scale well.

{{% vendor/name %}} [supports several different web servers](/languages/python/server/_index.md) for Python
```

> that you can use instead. The specific web server you choose will depend on your application and specific requirements.
> Let's look at how we can switch our project to use [gunicorn](https://gunicorn.org/) + PORT.

Reviewer: ```suggestion
that you can use instead. The specific web server you choose will depend on your application and specific requirements.

Let's look at how we can switch our project to use [gunicorn](https://gunicorn.org/) + PORT.
```

> Reopen the `{{< vendor/configfile "app" >}}` file. Locate the `web:commands:start` section where in the previous section
> we added `flask run -p $PORT`. We now want to change it to `gunicorn -w 4 'autoapp:app'`

Reviewer: ```suggestion
Reopen the `{{< vendor/configfile "app" >}}` file.

Locate the `web:commands:start` section where in the previous section
we added `flask run -p $PORT`. We now want to change it to `gunicorn -w 4 'autoapp:app'`
```

```yaml {configFile="app"}
    web:
      # Commands are run once after deployment to start the application process.
      commands:
        # The command to launch your app. If it terminates, it’s restarted immediately.
        # You can use the $PORT or the $SOCKET environment variable depending on the socket family of your upstream
        start: "gunicorn -w 4 'autoapp:app'"
```

Commit the changes and push them up to your environment:

```shell
$ git add {{< vendor/configfile "app" >}}
$ git commit -m "changes project to use gunicorn"

> $ {{% vendor/cli %}} e:push -y

Author: use either `upsun push` or `upsun environment:push` not the `e` shortcut.

Author: done

```

## Conclusion
Now that you have your Flask application up and running on {{% vendor/name %}}, we'll explore the different options you
have for a more robust local development environment, adding a source control integration, and adding
various services to your project. But for now, go forth and Deploy (even on Fridays)!

> {{< guide-buttons previous="Local Development" type="previous" >}}

Reviewer: ```suggestion
```

Author: @AnouckColson this is supposed to be part of the larger guide, and not as a standalone section, hence the button to go back to the previous. It's not really written to stand on its own and will need to be rewritten in order to do so. My initial plan was to later link to a section that describes _several_ alternate web server options on Upsun once those are built.

> Now that you have your Flask application up and running on {{% vendor/name %}}, we'll explore the different options you
> have for a more robust local development environment, adding a source control integration, and adding
> various services to your project. But for now, go forth and Deploy (even on Fridays)!
> 
> {{< guide-buttons previous="Local Development" type="previous" >}}

Reviewer: ```suggestion
have for a more robust [local development environment](/development/_index.md), [adding a source control integration](/integrations/_index.md), and [adding
various services](/add-services.md) to your project. But for now, go forth and Deploy (even on Fridays)!
```
